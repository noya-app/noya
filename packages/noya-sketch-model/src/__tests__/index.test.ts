import { SketchModel } from '../index';

jest.mock('noya-utils', () => {
  const uuid = () => '0';
  const original = jest.requireActual('noya-utils');
  return { ...original, uuid };
});

test('creates colors', () => {
  expect(SketchModel.color()).toMatchSnapshot();
});

test('creates gradients', () => {
  expect(SketchModel.gradient()).toMatchSnapshot();
});

test('creates graphicsContextSettings', () => {
  expect(SketchModel.graphicsContextSettings()).toMatchSnapshot();
});

test('creates borders', () => {
  expect(SketchModel.border()).toMatchSnapshot();
});

test('creates fills', () => {
  expect(SketchModel.fill()).toMatchSnapshot();
});

test('creates rectangles', () => {
  expect(SketchModel.rectangle()).toMatchSnapshot();
});

test('creates ovals', () => {
  expect(SketchModel.oval()).toMatchSnapshot();
});

test('creates bitmaps', () => {
  expect(SketchModel.bitmap()).toMatchSnapshot();
});

test('creates texts', () => {
  expect(SketchModel.text()).toMatchSnapshot();
});

test('creates shapePaths', () => {
  expect(SketchModel.shapePath()).toMatchSnapshot();
});

test('creates groups', () => {
  expect(SketchModel.group()).toMatchSnapshot();
});

test('creates artboards', () => {
  expect(SketchModel.artboard()).toMatchSnapshot();
});

test('creates symbolMasters', () => {
  expect(SketchModel.symbolMaster()).toMatchSnapshot();
});

test('creates symbolInstances', () => {
  expect(SketchModel.symbolInstance({ symbolID: '1' })).toMatchSnapshot();
});

test('creates metas', () => {
  expect(SketchModel.meta()).toMatchSnapshot();
});

test('creates documents', () => {
  expect(SketchModel.document()).toMatchSnapshot();
});
