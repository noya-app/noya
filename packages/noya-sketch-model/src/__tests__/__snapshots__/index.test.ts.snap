// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates artboards 1`] = `
Object {
  "_class": "artboard",
  "backgroundColor": Object {
    "_class": "color",
    "alpha": 1,
    "blue": 1,
    "green": 1,
    "red": 1,
  },
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "groupLayout": Object {
    "_class": "MSImmutableFreeformGroupLayout",
  },
  "hasBackgroundColor": false,
  "hasClickThrough": false,
  "hasClippingMask": false,
  "horizontalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
  "includeBackgroundColorInExport": true,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isFlowHome": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [],
  "name": "Artboard",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "presetDictionary": Object {},
  "resizesContent": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "verticalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
}
`;

exports[`creates bitmaps 1`] = `
Object {
  "_class": "bitmap",
  "booleanOperation": -1,
  "clippingMask": "{{0, 0}, {1, 1}}",
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "fillReplacesImage": false,
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClippingMask": false,
  "image": Object {
    "_class": "MSJSONOriginalDataReference",
    "_ref": "",
    "_ref_class": "MSImageData",
    "data": Object {
      "_data": "iVBORw0KGgoAAAANSUhEUgAAABgAAAAYAQMAAADaua+7AAAABlBMVEUAAAAAAAClZ7nPAAAAAXRSTlMAQObYZgAAABNJREFUCNdjYOD/TxL+/4GBFAwAvMsj3bQ3H74AAAAASUVORK5CYII=",
    },
    "sha1": Object {
      "_data": "e17bf8d6e2e327a621e67be822bb4352949bded3",
    },
  },
  "intendedDPI": 72,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "Image",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;

exports[`creates borders 1`] = `
Object {
  "_class": "border",
  "color": Object {
    "_class": "color",
    "alpha": 1,
    "blue": 0,
    "green": 0,
    "red": 0,
  },
  "contextSettings": Object {
    "_class": "graphicsContextSettings",
    "blendMode": 0,
    "opacity": 1,
  },
  "fillType": 0,
  "gradient": Object {
    "_class": "gradient",
    "elipseLength": 0,
    "from": "{0.5, 0}",
    "gradientType": 0,
    "stops": Array [
      Object {
        "_class": "gradientStop",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 1,
        },
        "position": 0,
      },
      Object {
        "_class": "gradientStop",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 0,
          "green": 0,
          "red": 0,
        },
        "position": 1,
      },
    ],
    "to": "{0.5, 1}",
  },
  "isEnabled": true,
  "position": 0,
  "thickness": 1,
}
`;

exports[`creates colors 1`] = `
Object {
  "_class": "color",
  "alpha": 1,
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;

exports[`creates documents 1`] = `
Object {
  "_class": "document",
  "assets": Object {
    "_class": "assetCollection",
    "colorAssets": Array [],
    "colors": Array [],
    "do_objectID": "0",
    "exportPresets": Array [],
    "gradientAssets": Array [],
    "gradients": Array [],
    "imageCollection": Object {
      "_class": "imageCollection",
      "images": Object {},
    },
    "images": Array [],
  },
  "colorSpace": 0,
  "currentPageIndex": 0,
  "do_objectID": "0",
  "documentState": Object {
    "_class": "documentState",
  },
  "fontReferences": Array [],
  "foreignLayerStyles": Array [],
  "foreignSwatches": Array [],
  "foreignSymbols": Array [],
  "foreignTextStyles": Array [],
  "layerStyles": Object {
    "_class": "sharedStyleContainer",
    "do_objectID": "0",
    "objects": Array [],
  },
  "layerSymbols": Object {
    "_class": "symbolContainer",
    "do_objectID": "0",
    "objects": Array [],
  },
  "layerTextStyles": Object {
    "_class": "sharedTextStyleContainer",
    "do_objectID": "0",
    "objects": Array [],
  },
  "pages": Array [],
  "sharedSwatches": Object {
    "_class": "swatchContainer",
    "do_objectID": "0",
    "objects": Array [],
  },
}
`;

exports[`creates fills 1`] = `
Object {
  "_class": "fill",
  "color": Object {
    "_class": "color",
    "alpha": 1,
    "blue": 0,
    "green": 0,
    "red": 0,
  },
  "contextSettings": Object {
    "_class": "graphicsContextSettings",
    "blendMode": 0,
    "opacity": 1,
  },
  "fillType": 0,
  "gradient": Object {
    "_class": "gradient",
    "elipseLength": 0,
    "from": "{0.5, 0}",
    "gradientType": 0,
    "stops": Array [
      Object {
        "_class": "gradientStop",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 1,
          "green": 1,
          "red": 1,
        },
        "position": 0,
      },
      Object {
        "_class": "gradientStop",
        "color": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 0,
          "green": 0,
          "red": 0,
        },
        "position": 1,
      },
    ],
    "to": "{0.5, 1}",
  },
  "isEnabled": true,
  "noiseIndex": 0,
  "noiseIntensity": 0,
  "patternFillType": 1,
  "patternTileScale": 1,
}
`;

exports[`creates gradients 1`] = `
Object {
  "_class": "gradient",
  "elipseLength": 0,
  "from": "{0.5, 0}",
  "gradientType": 0,
  "stops": Array [
    Object {
      "_class": "gradientStop",
      "color": Object {
        "_class": "color",
        "alpha": 1,
        "blue": 1,
        "green": 1,
        "red": 1,
      },
      "position": 0,
    },
    Object {
      "_class": "gradientStop",
      "color": Object {
        "_class": "color",
        "alpha": 1,
        "blue": 0,
        "green": 0,
        "red": 0,
      },
      "position": 1,
    },
  ],
  "to": "{0.5, 1}",
}
`;

exports[`creates graphicsContextSettings 1`] = `
Object {
  "_class": "graphicsContextSettings",
  "blendMode": 0,
  "opacity": 1,
}
`;

exports[`creates groups 1`] = `
Object {
  "_class": "group",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "groupLayout": Object {
    "_class": "MSImmutableFreeformGroupLayout",
  },
  "hasClickThrough": false,
  "hasClippingMask": false,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [],
  "name": "Group",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;

exports[`creates metas 1`] = `
Object {
  "app": "com.bohemiancoding.sketch3",
  "appVersion": "70.4",
  "autosaved": 0,
  "build": 109185,
  "commit": "1cee2bd6f09ef2258eb62d151bbe50dd6c3af3f2",
  "compatibilityVersion": 99,
  "created": Object {
    "app": "com.bohemiancoding.sketch3",
    "appVersion": "70.4",
    "build": 109185,
    "commit": "1cee2bd6f09ef2258eb62d151bbe50dd6c3af3f2",
    "compatibilityVersion": 99,
    "variant": "NONAPPSTORE",
    "version": 134,
  },
  "pagesAndArtboards": Object {},
  "saveHistory": Array [
    "NONAPPSTORE.109185",
  ],
  "variant": "NONAPPSTORE",
  "version": 134,
}
`;

exports[`creates ovals 1`] = `
Object {
  "_class": "oval",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClippingMask": false,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "Oval",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "points": Array [
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0.77614237490000004, 1}",
      "curveMode": 2,
      "curveTo": "{0.22385762510000001, 1}",
      "hasCurveFrom": true,
      "hasCurveTo": true,
      "point": "{0.5, 1}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{1, 0.22385762510000001}",
      "curveMode": 2,
      "curveTo": "{1, 0.77614237490000004}",
      "hasCurveFrom": true,
      "hasCurveTo": true,
      "point": "{1, 0.5}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0.22385762510000001, 0}",
      "curveMode": 2,
      "curveTo": "{0.77614237490000004, 0}",
      "hasCurveFrom": true,
      "hasCurveTo": true,
      "point": "{0.5, 0}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0, 0.77614237490000004}",
      "curveMode": 2,
      "curveTo": "{0, 0.22385762510000001}",
      "hasCurveFrom": true,
      "hasCurveTo": true,
      "point": "{0, 0.5}",
    },
  ],
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;

exports[`creates rectangles 1`] = `
Object {
  "_class": "rectangle",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "fixedRadius": 0,
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClippingMask": false,
  "hasConvertedToNewRoundCorners": true,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "Rectangle",
  "nameIsFixed": false,
  "needsConvertionToNewRoundCorners": false,
  "pointRadiusBehaviour": 1,
  "points": Array [
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0, 0}",
      "curveMode": 1,
      "curveTo": "{0, 0}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{0, 0}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{1, 0}",
      "curveMode": 1,
      "curveTo": "{1, 0}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{1, 0}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{1, 1}",
      "curveMode": 1,
      "curveTo": "{1, 1}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{1, 1}",
    },
    Object {
      "_class": "curvePoint",
      "cornerRadius": 0,
      "curveFrom": "{0, 1}",
      "curveMode": 1,
      "curveTo": "{0, 1}",
      "hasCurveFrom": false,
      "hasCurveTo": false,
      "point": "{0, 1}",
    },
  ],
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;

exports[`creates shapePaths 1`] = `
Object {
  "_class": "shapePath",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClippingMask": false,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "Path",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "points": Array [],
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
}
`;

exports[`creates symbolInstances 1`] = `
Object {
  "_class": "symbolInstance",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClippingMask": false,
  "horizontalSpacing": 0,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "name": "Symbol",
  "nameIsFixed": false,
  "overrideValues": Array [],
  "pointRadiusBehaviour": 1,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "scale": 1,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "symbolID": "1",
  "verticalSpacing": 0,
}
`;

exports[`creates symbolMasters 1`] = `
Object {
  "_class": "symbolMaster",
  "allowsOverrides": true,
  "backgroundColor": Object {
    "_class": "color",
    "alpha": 1,
    "blue": 1,
    "green": 1,
    "red": 1,
  },
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "groupLayout": Object {
    "_class": "MSImmutableInferredGroupLayout",
    "axis": 0,
    "layoutAnchor": 1,
    "maxSize": 0,
    "minSize": 0,
  },
  "hasBackgroundColor": false,
  "hasClickThrough": true,
  "hasClippingMask": false,
  "horizontalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
  "includeBackgroundColorInExport": false,
  "includeBackgroundColorInInstance": false,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isFlowHome": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [],
  "name": "Symbol",
  "nameIsFixed": false,
  "overrideProperties": Array [],
  "pointRadiusBehaviour": 1,
  "presetDictionary": Object {},
  "resizesContent": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "symbolID": "0",
  "verticalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
}
`;

exports[`creates texts 1`] = `
Object {
  "_class": "text",
  "attributedString": Object {
    "_class": "attributedString",
    "attributes": Array [
      Object {
        "_class": "stringAttribute",
        "attributes": Object {
          "MSAttributedStringColorAttribute": Object {
            "_class": "color",
            "alpha": 1,
            "blue": 0,
            "green": 0,
            "red": 0,
          },
          "MSAttributedStringFontAttribute": Object {
            "_class": "fontDescriptor",
            "attributes": Object {
              "name": "Roboto",
              "size": 18,
            },
          },
          "paragraphStyle": Object {
            "_class": "paragraphStyle",
            "alignment": 0,
          },
          "textStyleVerticalAlignmentKey": 0,
        },
        "length": 0,
        "location": 0,
      },
    ],
    "string": "",
  },
  "automaticallyDrawOnUnderlyingPath": false,
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "0",
  "dontSynchroniseWithSymbol": false,
  "edited": false,
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "glyphBounds": "{{0, 5}, {126, 17}}",
  "hasClippingMask": false,
  "isClosed": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "lineSpacingBehaviour": 3,
  "name": "Text",
  "nameIsFixed": false,
  "pointRadiusBehaviour": 1,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "blur": Object {
      "_class": "blur",
      "center": "{0.5, 0.5}",
      "isEnabled": false,
      "motionAngle": 0,
      "radius": 10,
      "saturation": 1,
      "type": 0,
    },
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": true,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "0",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "textStyle": Object {
      "_class": "textStyle",
      "encodedAttributes": Object {
        "MSAttributedStringColorAttribute": Object {
          "_class": "color",
          "alpha": 1,
          "blue": 0,
          "green": 0,
          "red": 0,
        },
        "MSAttributedStringFontAttribute": Object {
          "_class": "fontDescriptor",
          "attributes": Object {
            "name": "Roboto",
            "size": 18,
          },
        },
        "paragraphStyle": Object {
          "_class": "paragraphStyle",
          "alignment": 0,
        },
        "textStyleVerticalAlignmentKey": 0,
      },
      "verticalAlignment": 0,
    },
    "windingRule": 1,
  },
  "textBehaviour": 0,
}
`;
